### PR TITLE
Statistics: add support for percentage unit in addition to time

### PR DESCRIPTION
**Pull request by LebedevRI**
I think, `stddev` statistic is useful, but confusing.

What does it mean if `stddev` of `1ms` is reported?
Is that good or bad? If the `median` is `1s`,
then that means that the measurements are pretty noise-less.

And what about `stddev` of `100ms` is reported?
If the `median` is `1s` - awful, if the `median` is `10s` - good.

And hurray, there is just the statistic that we need:
https://en.wikipedia.org/wiki/Coefficient_of_variation

But, naturally, that produces a value in percents,
but the statistics are currently hardcoded to produce time.

So this refactors thinkgs a bit, and allows a percentage unit for statistics.

I'm not sure whether or not `benchmark` would be okay
with adding this `RSD` statistic by default,
but regales, that is a separate patch.

Refs. https://github.com/google/benchmark/issues/1146

### CODE REVIEW

File: include/benchmark/benchmark.h [hunk 1]
```diff
@@ -450,6 +450,8 @@ enum BigO { oNone, o1, oN, oNSquared, oNCubed, oLogN, oNLogN, oAuto, oLambda };
 
 typedef uint64_t IterationCount;
 
+enum StatisticUnit { Time, Percentage };
```
Comment (dominichamon) [inline include/benchmark/benchmark.h:453]: https://google.github.io/styleguide/cppguide.html#Enumerator_Names

`kTime` and `kPercentage` please

File: src/console_reporter.cc [hunk 2]
```diff
@@ -142,10 +142,16 @@ void ConsoleReporter::PrintRunData(const Run& result) {
...
-  } else {
+  } else if (result.run_type != Run::RT_Aggregate ||
+             result.aggregate_unit == StatisticUnit::Time) {
     const char* timeLabel = GetTimeUnitString(result.time_unit);
     printer(Out, COLOR_YELLOW, "%s %-4s %s %-4s ", real_time_str.c_str(), timeLabel,
             cpu_time_str.c_str(), timeLabel);
+  } else {
+    assert(result.aggregate_unit == StatisticUnit::Percentage);
+    printer(Out, COLOR_YELLOW, "%10.2f %-4s %10.2f %-4s ",
+            (100. * result.real_accumulated_time), "%",
```
Comment (dominichamon) [inline src/console_reporter.cc:153]: bit awkward this is called `_time` and now isn't always one. maybe a rename? or is that too intrusive?

File: src/statistics.cc [hunk 3]
```diff
@@ -167,13 +168,15 @@ std::vector<BenchmarkReporter::Run> ComputeStats(
...
-    // We will divide these times by data.iterations when reporting, but the
-    // data.iterations is not nessesairly the scale of these measurements,
-    // because in each repetition, these timers are sum over all the iterations.
-    // And if we want to say that the stats are over N repetitions and not
-    // M iterations, we need to multiply these by (N/M).
-    data.real_accumulated_time *= iteration_rescale_factor;
-    data.cpu_accumulated_time *= iteration_rescale_factor;
+    if (data.aggregate_unit == StatisticUnit::Time) {
+      // We will divide these times by data.iterations when reporting, but the
+      // data.iterations is not nessesairly the scale of these measurements,
```
Comment (dominichamon) [inline src/statistics.cc:173]: while you're here: "necessarily"

**Review by dominichamon — COMMENTED** [review]
i always thought that alongside median we should have other percentiles, instead of mean and stddev, because this gives you a better idea of the distribution of results. maybe the CV is a good middle ground instead of arguing about which percentiles to show.

i'd be supportive of replacing stddev with stddev/mean (and we calculate both anyway...)

File: src/console_reporter.cc (hunk 2 above)
Comment (LebedevRI) [reply src/console_reporter.cc:153]: I keep delaying dealing with this, so i still don't want to change this now,
but this should be a vector of measurements, just two hardcoded "timers" isn't right.

File: src/statistics.cc (hunk 3 above)
Comment (LebedevRI) [reply src/statistics.cc:173]: If only Qt/Qtcreator would finally add spell checking support (:

**Comment by LebedevRI** [comment]
@dominichamon thank you for taking a look!

> i always thought that alongside median we should have other percentiles, instead of mean and stddev, because this gives you a better idea of the distribution of results. maybe the CV is a good middle ground instead of arguing about which percentiles to show.

Aha, sounds good to me.

> i'd be supportive of replacing stddev with stddev/mean (and we calculate both anyway...)
I'm not really sure about completely replacing it, but then we also don't want to bloat the output

with too many enabled-by-default statistics.

**Pull request merged**